docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+    controller-tools.k8s.io: "1.0"
+  name: directimagemigration-sample
+spec:
+  srcMigClusterRef:
+    name: migcluster-local
+    namespace: openshift-migration
+
+  destMigClusterRef:
+    name: migcluster-remote
+    namespace: openshift-migration
+
+  namespaces:
+  - nginx-example
+```
+
+-   spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the destination mig cluster where the operator is installed.
+
+-   spec.namespaces:  A list of all namespaces to run DIM to get all the imagestreams in these namespaces.
+
+-   spec.srcMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the operator is installed.
+
+## Direct ImageStream Migration
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: DirectImageStreamMigration
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: directimagestreammigration-sample
+spec:
+  srcMigClusterRef:
+    name: migcluster-local
+    namespace: openshift-migration
+
+  destMigClusterRef:
+    name: migcluster-remote
+    namespace: openshift-migration
+
+  imageStreamRef:
+    name: nginx
+    namespace: nginx-example
+
+  destNamespace: new-namespace
+  ```
+
+- spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the destination mig cluster where the operator is installed.
+
+- spec.destNamespace: A string holding the name of the namespace on destination cluster where imagestreams should be migrated.
+
+- spec.imageStreamRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the imagesstream is present.
+
+
+- spec.srcMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the operator is installed.
+
+     
+## Direct Volume Migration
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: DirectVolumeMigration
+metadata:
+  name: direct1
+  namespace: openshift-migration
+spec:
+  createDestinationNamespaces: false
+  deleteProgressReportingCRs: false
+  destMigClusterRef:
+    name: host
+    namespace: openshift-migration
+  persistentVolumeClaims:
+  - name: pvc-0
+    namespace: pvc-migrate-bmark-1
+  - name: pvc-0
+    namespace: pvc-migrate-bmark-2
+  - name: pvc-0
+    namespace: pvc-migrate-bmark-3
+  - name: pvc-1
+    namespace: pvc-migrate-bmark-3
+  srcMigClusterRef:
+    name: ocp3
+    namespace: openshift-migration
+```
+
+- spec.createDestinationNamespaces: A boolean flag that is set True to create namespaces in destination cluster.
+
+- spec.deleteProgressReportingCRs: A boolean flag to specify if progress reporting CRs needs to be deleted or not.
```

Comment:
@eriknelson @shubham-pampattiwar 
What are `progress reporting CRs`?

Reply:
@apinnick A `DirectVolumeMigrationProgress` CR or a DVMP CR is created when a user migrates persistent volumes via the Direct Volume Migration option. This CR is created per PV involved in the migration and it helps in monitoring the migration progress of that particular PV, provides granular details like transfer rate, age etc.

Reply:
@shubham-pampattiwar  Why would a user want to delete or not delete the `DirectVolumeMigrationProgress` CR?

Reply:
@apinnick The progress reporting CRs have the log messages of failure. So by default, they are left around. If the user wants to delete them, they would have to turn this flag to true.